compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -256,7 +256,7 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisTrans
         case tree @ Annotated(annotated, annot) =>
           cpy.Annotated(tree)(transform(annotated), transformAnnot(annot))
         case tree: AppliedTypeTree =>
-          Checking.checkAppliedType(tree)
+          Checking.checkAppliedType(tree, boundsCheck = !ctx.mode.is(Mode.Pattern))
```

Comment:
Now I'm confusing about why we would write this instead of `if (!ctx.mode.is(Mode.Pattern)) Checking.checkAppliedType(tree)` :)

Reply:
Because checkAppliedType also does other things besides bounds checking.